types/part_set_test.go

```diff
@@ -63,10 +66,64 @@ func TestBasicPartSet(t *testing.T) {
 	assert.Equal(t, data, data2)
 }
 
+func TestEncodingDecodingRoundTrip(t *testing.T) {
```

Comment:
in a future PR, it would be good to add a test for the case where the last chunk is not complete to see if the padding works as expected

Reply:
we will need to add another field to the compact block to know where to slice